libs/libcommon/include/common/logger_useful.h

```diff
@@ -4,41 +4,39 @@
 
 #include <sstream>
 #include <Poco/Logger.h>
+#include <Poco/Message.h>
+#include <Poco/Version.h>
+#include <Core/SettingsCommon.h>
+#include <Common/CurrentThread.h>
 
 #ifndef QUERY_PREVIEW_LENGTH
 #define QUERY_PREVIEW_LENGTH 160
 #endif
 
 using Poco::Logger;
+using Poco::Message;
+using DB::LogsLevel;
+using DB::CurrentThread;
 
 /// Logs a message to a specified logger with that level.
 
-#define LOG_TRACE(logger, message) do { \
-    if ((logger)->trace()) {\
-    std::stringstream oss_internal_rare;    \
-    oss_internal_rare << message; \
-    (logger)->trace(oss_internal_rare.str());}} while(false)
-
-#define LOG_DEBUG(logger, message) do { \
-    if ((logger)->debug()) {\
-    std::stringstream oss_internal_rare;    \
-    oss_internal_rare << message; \
-    (logger)->debug(oss_internal_rare.str());}} while(false)
-
-#define LOG_INFO(logger, message) do { \
-    if ((logger)->information()) {\
-    std::stringstream oss_internal_rare;    \
-    oss_internal_rare << message; \
-    (logger)->information(oss_internal_rare.str());}} while(false)
-
-#define LOG_WARNING(logger, message) do { \
-    if ((logger)->warning()) {\
-    std::stringstream oss_internal_rare;    \
-    oss_internal_rare << message; \
-    (logger)->warning(oss_internal_rare.str());}} while(false)
-
-#define LOG_ERROR(logger, message) do { \
-    if ((logger)->error()) {\
-    std::stringstream oss_internal_rare;    \
-    oss_internal_rare << message; \
-    (logger)->error(oss_internal_rare.str());}} while(false)
+#define LOG_SIMPLE(logger, message, priority, PRIORITY) do                                      \
+{                                                                                               \
+    const bool is_clients_log = (CurrentThread::getGroup() != nullptr) &&                       \
+            (CurrentThread::getGroup()->client_logs_level >= (priority));                       \
+    if ((logger)->is((PRIORITY)) || is_clients_log) {                                           \
+        std::stringstream oss_internal_rare;                                                    \
+        oss_internal_rare << message;                                                           \
+        if ( auto channel = (logger)->getChannel() ) {                                          \
+            channel->log(Message("", oss_internal_rare.str(), (PRIORITY)));                     \
```

Comment:
shouldn't first parameter (source) for message constructor be something like `(logger)->name()` ?

Reply:
Yes.

Reply:
I will commit It tomorrow with integration test.

Reply:
Also it would be cool to check explicitly the scenario: server has (for example) 'error' log level. client request 'trace', trace messages came to the client BUT NOT written to the server logs (I think it should work correctly right now, but would be good to have that check).

Reply:
It's Ok if these messages will also be written in server logs.
(better if we can avoid it but it's not necessary)

Reply:
Hm. I think it's better to avoid that, as if somebody decrease the error level for the server he made it for some reason - like avoiding noisy messages to see more impornatnt, better control on logs disk space, or existence of monitoring/alerting system which can react on the appearance of the big number of entries in the log file.  

I thought it will 'just work', but now i'm not sure (it looks like proper levels are not setup for separate channels in Loggers.cpp).

Reply:
Good enough.